project/src/main/scala/org/broadinstitute/clio/integrationtest/ClioIntegrationTestSettings.scala

```diff
@@ -1,42 +1,90 @@
-  val testDockerJenkins: TaskKey[Unit] =
-    taskKey[Unit]("Run clio integration tests via docker in Jenkins.")
-
-  private def creatTestTask(jenkinsTest: Boolean) = {
-    Def.task {
-      // Use the settings classDirectory for tests, and the project version
-      val testClassesDirectory = (classDirectory in Test).value
-      val clioVersion = version.value
-
-      // Wait for task test:compile, and the task to create the logs
-      (compile in Test).value
-      val log = streams.value.log
-
-      new ClioIntegrationTestRunner(testClassesDirectory,
-                                    clioVersion,
-                                    jenkinsTest,
-                                    log)
-        .run()
-    }
+object ClioIntegrationTestSettings extends ClioIntegrationTestKeys {
+
+  /** Directory to which all container logs will be sent during Dockerized integration tests. */
+  lazy val itLogDir: Initialize[File] = Def.setting {
+    target.value / "integration-test" / "logs"
+  }
+
+  /** Task to clear out the IT log dir before running tests. */
+  lazy val resetItLogs: Initialize[Task[File]] = Def.task {
+    val logDir = itLogDir.value
+    IO.delete(logDir)
+    IO.createDirectory(logDir)
+    /*
+     * Our Dockerized integration tests tail the clio log to check
+     * for the "started" message before starting tests, so we ensure
+     * it exists here.
+     */
+    val clioLog = logDir / "clio-server" / "clio.log"
+    IO.touch(clioLog)
+    clioLog
+  }
+
+  /** Environment variables to inject when running integration tests. */
+  lazy val itEnvVars: Initialize[Task[Map[String, String]]] = Def.task {
+    val logDir = itLogDir.value
+    val confDir = (classDirectory in IntegrationTest).value / "org" / "broadinstitute" / "clio" / "integrationtest"
+
+    Map(
+      "CLIO_DOCKER_TAG" -> version.value,
+      // TODO: Pull this version from elsewhere in the build.
+      "ELASTICSEARCH_DOCKER_TAG" -> "5.4.0_6",
+      "LOG_DIR" -> logDir.getAbsolutePath,
+      "CONF_DIR" -> confDir.getAbsolutePath
+    )
   }
 
   /** Settings to add to the project */
-  lazy val settings: Seq[Setting[Task[Unit]]] = Seq(
-    testDocker := creatTestTask(false).value,
-    testDockerJenkins := creatTestTask(true).value)
+  lazy val settings: Seq[Setting[_]] = {
+
+    Seq.concat(
+      Defaults.itSettings,
+      Seq(
+        /*
+         * Override the top-level `test` definition to be a no-op so running "sbt test"
+         * won't trigger integration tests along with the unit tests in other projects.
+         *
+         * Integration tests should be explicitly run with the keys defined below, or "it:test".
+         */
+        test := {},
+        /*
+         * Define aliases for "it:testOnly XYZ" to make running integration tests against
+         * different setups / environments more convenient.
+         */
+        testDocker := (testOnly in IntegrationTest)
+          .toTask(" *FullDockerIntegrationSpec")
```

Comment:
It does depend on the class name, but I don't think it can use a type reference (if I understand what you mean). The build code is compiled as an entirely separate project from the app code within SBT, and the two run binary incompatible versions of Scala (SBT is stuck on 2.10). If we really wanted to lock down one location for the spec names across the build and app code I think we'd need to auto-generate the source files from SBT with the expected names, and I'm not sure we want to head down that road.

At the least, I'll add big warning comments above the test classes saying that if they're renamed, these task definitions also need to be modified.

Alternatively I could just scrap these tasks. Currently, they're just serving as aliases for the `it:testOnly` input task that SBT provides, so anyone who's gotten the muscle memory of typing `testDocker` doesn't have to change their workflow. If we don't think they're worth it I can toss them out, but I think they're a nice convenience.